fuzzing/tools/launcher.py

```diff
@@ -18,58 +18,85 @@
 behind a number of arbitrary fuzzing engines.
 """
 
+import os
+import shutil
+import sys
+
 from absl import app
 from absl import flags
-import os
 
 FLAGS = flags.FLAGS
 
+flags.DEFINE_string(
+    "engine_launcher", None,
+    "Path to a shell script that launches the fuzzing engine executable with the appropriate command line arguments."
+)
+
+flags.DEFINE_string("binary_path", None, "Path to the fuzz test binary.")
+
 flags.DEFINE_bool(
     "regression", False,
     "If set True, the script will trigger the target as a regression test.")
 
-flags.DEFINE_enum(
-    "engine", "default", ["default", "libfuzzer"],
-    "The type of the engine, the default is to run a gUnit test.")
-
 flags.DEFINE_integer(
     "timeout_secs",
     0,
     "The maximum duration, in seconds, of the fuzzer run launched.",
     lower_bound=0)
 
-flags.DEFINE_list(
-    "fuzzer_extra_args", None,
-    "If non-empty, the elements will be passed to the fuzzer as arguments.")
-
 flags.DEFINE_string(
-    "corpus_dir", "",
+    "corpus_dir", None,
     "If non-empty, a directory that will be used as a seed corpus for the fuzzer."
 )
 
-flags.DEFINE_string("dict", "",
+flags.DEFINE_string("dictionary_path", None,
                     "If non-empty, a dictionary file of input keywords.")
 
+flags.DEFINE_string(
+    "fuzzing_output_root", "/tmp/fuzzing",
```

Comment:
as an aside, do you know if there's a bazel "best practice" for writing outputs? should they completely exist in the bazel cache/bazel-out directory (or is this where it ends up going to since it's executed within the bazel process)

Reply:
As far as I know, Bazel provides a [`TEST_TMPDIR` variable](https://docs.bazel.build/versions/master/test-encyclopedia.html#test-interaction-with-the-filesystem) that the test can use to write private, ephemeral data. But this would get cleaned up after each test execution, and I'm thinking that developers might want to reuse the generated corpus in between runs, in case the fuzz target gets interrupted.

For this "persistent" data, I'm not aware of any facility offered by Blaze. I imagine using a `/tmp` path that maps to a unique path per fuzz target would be enough. This is what the TODO below is about, since we currently cobble all the corpus output under a single directory across all fuzz target runs.

Reply:
Yeah, I agree holding on to the generated corpus is useful. I imagine this is OK, de-clobbering sounds good to me.